lacework/resource_lacework_integration_gcp_gke_audit_log.go

```diff
@@ -0,0 +1,305 @@
+		Importer: &schema.ResourceImporter{State: importLaceworkIntegration},
+	}
+}
+
+var gcpGkeAuditLogIntegrationSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The integration name.",
+	},
+	"intg_guid": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"enabled": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "The state of the external integration.",
+	},
+	"retries": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Default:     5,
+		Description: "The number of attempts to create the external integration.",
+	},
+	"credentials": {
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"client_id": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"private_key_id": {
+					Type:      schema.TypeString,
+					Required:  true,
+					Sensitive: true,
+					DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+						// @afiune we can't compare this element since our API, for security reasons,
+						// does NOT return the private key configured in the Lacework server. So if
+						// any other element changed from the credentials then we trigger a diff
+						return !d.HasChanges(
+							"name", "integration_type", "project_id", "organization_id",
+							"subscription", "enabled",
+							"credentials.0.client_id",
+							"credentials.0.client_email",
+						)
+					},
+				},
+				"client_email": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"private_key": {
+					Type:      schema.TypeString,
+					Required:  true,
+					Sensitive: true,
+					DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+						// @afiune we can't compare this element since our API, for security reasons,
+						// does NOT return the private key configured in the Lacework server. So if
+						// any other element changed from the credentials then we trigger a diff
+						return !d.HasChanges(
+							"name", "integration_type", "project_id", "organization_id",
+							"subscription", "enabled",
+							"credentials.0.client_id",
+							"credentials.0.client_email",
+						)
+					},
+				},
+			},
+		},
+	},
+	"integration_type": {
+		Type:     schema.TypeString,
+		Required: true,
+		StateFunc: func(val interface{}) string {
+			return strings.ToUpper(val.(string))
+		},
+		ValidateFunc: func(value interface{}, key string) ([]string, []error) {
+			switch strings.ToUpper(value.(string)) {
+			case "PROJECT", "ORGANIZATION":
+				return nil, nil
+			default:
+				return nil, []error{
+					fmt.Errorf("%s: can only be either 'PROJECT' or 'ORGANIZATION'", key),
+				}
+			}
+		},
+	},
+	"organization_id": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "The GCP Organization ID (required when integration_type is organization).",
```

Comment:
If this field is required when the integration type is `ORGANIZATION`, could we add a validation that checks it and returns a helpful message to the end-user?

Reply:
@afiune. Do we want to catch this at this level or should we let the API return an error on this? Generally I see Terraform doing the latter and letting the API do the validation.

Reply:
I am ok leaving it to the API, though, we try to add validation in resources so that the user detects these kinds of things early on (that is, during `terraform plan`) rather than in the middle of an apply.

Up to you to decide what's best for our users here.

Reply:
Sure I'll see what I can do in terms of validation at this level. Looks like `CustomizeDiff` might give us what we need to validate this.